vignettes/contribution.Rmd

```diff
@@ -0,0 +1,371 @@
+---
+title: "Contributing Guide for Developers"
```

Comment:
As described in the issue, let's differentiate between
1) this vignette, describing the package structure and
2) the contributing guide, which is located at `.github/CONTRIBUTING.md`, so it exists already but should be extended, see issue description

for this vignette please change title to e.g. "Code Structure" or "Package Structure" and adapt file name accordingly.